raxx_elli/lib/raxx/elli/handler.ex

```diff
@@ -30,10 +30,19 @@ defmodule Raxx.Elli.Handler do
     headers = :elli_request.headers(elli_request) |> Enum.map(fn ({k, v}) ->
       {String.downcase(k), String.downcase(v)}
     end)
-    {"Host", authority} = :elli_request.headers(elli_request) |> List.keyfind("Host", 0)
-    [host, port] = String.split(authority, ":")
+
+	[host, port] = 
+	case :elli_request.headers(elli_request) |> List.keyfind("Host", 0) do
+	  {"Host", authority} ->
+		case String.split(authority, ":") do
+			[host, port] -> [host, port]
+			[host] -> [host, "80"] # we need to set the correct port some how...
```

Comment:
Is this not the same as finding the correct scheme. if http default port can assume to be 80 if https can assumed to be 443 (I think?)